tools/SourceKit/tools/sourcekitd-test/CMakeLists.txt

```diff
@@ -8,17 +8,15 @@ else()
   set(SOURCEKITD_TEST_LINK_LIBS sourcekitd)
 endif()
 
-if(SOURCEKIT_NEED_EXPLICIT_LIBDISPATCH)
-  set(SOURCEKITD_TEST_LINK_LIBS ${SOURCEKITD_TEST_LINK_LIBS} dispatch)
-endif()
-
 add_sourcekit_executable(sourcekitd-test
   sourcekitd-test.cpp
   TestOptions.cpp
-  LINK_LIBS ${SOURCEKITD_TEST_LINK_LIBS} SourceKitSupport
-    clangRewrite clangLex clangBasic
+  LINK_LIBS SourceKitSupport clangRewrite clangLex clangBasic
   LLVM_COMPONENT_DEPENDS core support option coverage lto
 )
+if(SOURCEKIT_NEED_EXPLICIT_LIBDISPATCH)
+  target_link_libraries(sourcekitd-test PRIVATE dispatch BlocksRuntime)
```

Comment:
Is this necessary ? `sourcekitd-test` depends on `SourceKitSupport`, which also gets those dependencies.

Reply:
Hmm, yeah, it should be possible to clean that up, but I would prefer to do that in a follow up.